scheduler/scheduler-server/src/test/java/performancetests/recovery/NodeRecoveryTest.java

```diff
@@ -25,37 +25,117 @@
  */
 package performancetests.recovery;
 
-import org.apache.jmeter.protocol.java.sampler.AbstractJavaSamplerClient;
-import org.apache.jmeter.protocol.java.sampler.JavaSamplerContext;
-import org.apache.jmeter.samplers.SampleResult;
-import org.apache.jmeter.threads.JMeterContextService;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
-import performancetests.recovery.helper.NodeRecoveryHelper;
+import java.util.*;
 
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.objectweb.proactive.core.config.ProActiveConfiguration;
+import org.ow2.proactive.resourcemanager.common.RMConstants;
+import org.ow2.proactive.resourcemanager.core.RMCore;
+import org.ow2.proactive.resourcemanager.core.properties.PAResourceManagerProperties;
+import org.ow2.proactive.resourcemanager.frontend.ResourceManager;
 
-public class NodeRecoveryTest extends AbstractJavaSamplerClient {
+import functionaltests.nodesrecovery.RecoverInfrastructureTestHelper;
+import functionaltests.utils.*;
+import performancetests.helper.LogProcessor;
 
-    @Override
-    public SampleResult runTest(JavaSamplerContext javaSamplerContext) {
-        SampleResult sampleResult = null;
-        try {
-            final Integer nodesNumber = Integer.valueOf(JMeterContextService.getContext()
-                                                                            .getVariables()
-                                                                            .get("nodesNumber"));
 
-            NodeRecoveryHelper nodeRecoveryHelper = new NodeRecoveryHelper();
+/**
+ * @author ActiveEon Team
+ * @since 01/12/17
+ */
+@RunWith(Parameterized.class)
+public class NodeRecoveryTest extends SchedulerFunctionalTestWithCustomConfigAndRestart {
+
+    static final String RM_CONFIGURATION_START = NodeRecoveryTest.class.getResource("/performancetests/config/rm-start.ini")
+                                                                       .getPath();
+
+    static final String RM_CONFIGURATION_RESTART = NodeRecoveryTest.class.getResource("/performancetests/config/rm-restart.ini")
+                                                                         .getPath();
+
+    @Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] { { 5, 1000 }, { 100, 2000 }, { 200, 1000 }, { 300, 3000 } });
```

Comment:
Since the data set is the same for `JobRecoveryTest` (and because it is interesting to always test both JobRecoveryTest and NodeRecoveryTest on the same data set), maybe we could share this data set through a dedicated class, for example RecoveryDataSet?

Reply:
With JUnit5 it will be the same csv file who will parametrize tests. I do not want to create additional class for that so I will make public static method and so JobRecovery class will depende on NodeRecovery class

Reply:
Wait, but time limits will be different for job and node recovery, isn't it?

Reply:
Yes, you are right the time limit will be different.